### PR TITLE
Adding the MlNetMklDeps package, adding the entry point, tests and docs for OLS.

### PR DESCRIPTION
**Pull request by sfilipi**
MKL package has been published

### CODE REVIEW

**Comment by codemzs** [comment]
Have you tested MKL works as expected?

File: pkg/MlNetMklDeps/MlNetMklDeps.nupkgproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk" DefaultTargets="Pack">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <PackageDescription>Intel(R) MKL dependencies for ML.Net.</PackageDescription>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../Microsoft.ML/Microsoft.ML.nupkgproj" />
```
Comment (Ivanidzo4ka) [inline pkg/MlNetMklDeps/MlNetMklDeps.nupkgproj]: Can we write down how exactly it's gonna work?
How it looks in my head based on conversations (can be far away from reality)
1. Have a internal repo with pre build custom dlls for each OS.
2. Publish them to nuget.org
3. Reference them in Microsoft.ML nuget as dependency

this steps have nothing to do with this package file, so I obviously don't understand something.

File: build/Dependencies.props [hunk 2]
```diff
@@ -8,5 +8,6 @@
     <SystemReflectionEmitLightweightPackageVersion>4.3.0</SystemReflectionEmitLightweightPackageVersion>
     <PublishSymbolsPackageVersion>1.0.0-beta-62824-02</PublishSymbolsPackageVersion>
 	<LightGBMPackageVersion>2.1.2.2</LightGBMPackageVersion>
+	<MlNetMklDepsPackageVersion>0.0.0.1</MlNetMklDepsPackageVersion>
```
Comment (Ivanidzo4ka) [inline build/Dependencies.props]: >	 [](start = 0, length = 1)

ze evil tabs!

**Comment by TomFinley** [comment]
Hi @sfilipi thanks for getting this started.

I see we have a package definition added, I think. But I might have expected the package definition to live somewhere else in a different repository somehow (since it cannot be part of the build here), and for components in ML.Net that need it to simply reference that externally defined nuget. Is that unfair?

I also feel like I might understand this better if we saw how existing component (like OLS) or upcoming ones (like #556) will actually consume this.

**Comment by codemzs** [comment]
Hi @tfinley@gmail.com ,I have tested MKL binaries by linking them with native SymSGD code in PR556 (that iteration is not posted yet) using CMAKE. The way we would do is by restoring MKL nuget before CMAKE is called by adding an external project to the sln file that package references MKL nuget, this step fetches the binaries to a known location on the drive and then we can link against the binaries in our CMAKE file. 

[https://github.com/dotnet/machinelearning/blob/8cfa2ed503c1cca78972d79b393fbffe13bde173/build.proj#L23](https://github.com/dotnet/machinelearning/blob/8cfa2ed503c1cca78972d79b393fbffe13bde173/build.proj#L23) (this line is executed before BuildNative is called)


---
In reply to: [407535938](https://github.com/dotnet/machinelearning/pull/578#issuecomment-407535938) [](ancestors = 407535938)

File: src/Microsoft.ML.StandardLearners/Standard/doc.xml [hunk 3]
```diff
@@ -68,6 +68,26 @@
...
+        Train an OLS regression model.
+      </summary>
+      <remarks>
+        <a href='https://en.wikipedia.org/wiki/Ordinary_least_squares'>Ordinary least squares (OLS)</a> is a parameterized regression method. 
+        It assumes that the conditional mean of the dependent variable follows a linear function of the dependent variables.
+        The parameters of the regressor can be estimated by minimizing the squares of the difference between observed values and the predictions.
+      </remarks>
+       <example>
+        <code language="csharp">
+          new StochasticDualCoordinateAscentRegressor
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardLearners/Standard/doc.xml]: > new StochasticDualCoordinateAscentRegressor [](start = 9, length = 44)

I'll fix this.

**Comment by sfilipi** [comment]
The changes went in through the other PR.

**Pull request closed without merging**